Tests/NIOPosixTests/EventLoopFutureIsolatedTests.swift

```diff
@@ -0,0 +1,354 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import NIOPosix
+import XCTest
+
+final class SuperNotSendable {
+    var x: Int = 5
+}
+
+@available(*, unavailable)
+extension SuperNotSendable: Sendable {}
+
+final class EventLoopFutureIsolatedTest: XCTestCase {
+    func testCompletingPromiseWithNonSendableValue() throws {
+        let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer {
+            try! group.syncShutdownGracefully()
+        }
```

Comment:
Not that it matters much, but should we prefer singletons in tests as they save a little boilerplate.